lib/galaxy/webapps/galaxy/buildapp.py

```diff
@@ -76,6 +76,10 @@ def paste_app_factory(global_conf, **kwargs):
     # Force /activate to go to the controller
     webapp.add_route('/activate', controller='user', action='activate')
 
+    # OAuth2.0 endpoints.
+    webapp.add_route('/oauth2authentication/google', controller='oauth2', action='google_authn')
+    webapp.add_route('/oauth2callback/google', controller='oauth2', action='google_callback')
```

Comment:
If the providers are loaded by this point in time (and attached to something like `app`) you could probably iterate over those and construct these routes programmatically from configured providers which might be more flexible should a site add a new auth provider.